app/scripts/lib/createRPCMethodTrackingMiddleware.js

```diff
@@ -193,25 +189,21 @@ export default function createRPCMethodTrackingMiddleware({
           );
 
           if (securityProviderResponse?.flagAsDangerous === 1) {
-            properties.ui_customizations = ['flagged_as_malicious'];
+            eventProperties.ui_customizations = [
+              METAMETRIC_KEY_OPT.ui_customizations.flaggedAsMalicious,
+            ];
           } else if (securityProviderResponse?.flagAsDangerous === 2) {
-            properties.ui_customizations = ['flagged_as_safety_unknown'];
-          } else {
-            properties.ui_customizations = null;
+            eventProperties.ui_customizations = [
+              METAMETRIC_KEY_OPT.ui_customizations.flaggedAsSafetyUnknown,
+            ];
           }
 
           if (method === MESSAGE_TYPE.PERSONAL_SIGN) {
             const { isSIWEMessage } = detectSIWE({ data });
             if (isSIWEMessage) {
-              properties.ui_customizations === null
-                ? (properties.ui_customizations = [
-                    METAMETRIC_KEY_OPTIONS[METAMETRIC_KEY.UI_CUSTOMIZATIONS]
-                      .SIWE,
-                  ])
-                : properties.ui_customizations.push(
-                    METAMETRIC_KEY_OPTIONS[METAMETRIC_KEY.UI_CUSTOMIZATIONS]
-                      .SIWE,
-                  );
```

Comment:
turns out Array.push returns the length of the mutated array. this is a JS gotchya. fixed by using Array.concat

bug introduced #17688